packages/plugin-ngrok/src/tunnel.test.ts

```diff
@@ -93,16 +95,18 @@ describe('start', () => {
     vi.mocked(os.platformAndArch).mockReturnValue({platform: 'windows', arch: 'arch'})
 
     // When
-    await hookStart(port)
+    const got = await hookStart(port)
 
     // Then
-    expect(error.Abort).toHaveBeenCalledWith(
+    expect(AbortError).toHaveBeenCalledWith(
       'The ngrok tunnel could not be started.\n\nerror message contains code err_ngrok_108',
       expect.stringContaining(
         'Kill all the ngrok processes with \u001b[1m\u001b[33mtaskkill /f /im ngrok.exe\u001b[39m\u001b[22m',
       ),
     )
-    expect(renderFatalError).toHaveBeenCalledWith(expect.any(error.Abort))
+    expect(renderFatalError).toHaveBeenCalledWith(expect.any(AbortError))
+    expect(got.isErr() && got.error.type).toEqual('tunnel-already-running')
+    expect(got.isErr() && got.error.message).toEqual('error message contains code err_ngrok_108')
```

Comment:
Nitpick:

```js
 expect(got.error?.type).toEqual('tunnel-already-running')
 expect(got.error?.message).toEqual('error message contains code err_ngrok_108')
```

The presence of an error in `got.error` implicitly tells that the result represents an error so there's no need for `got.isErr()`

Reply:
Oh I see! Today I learned about [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates). It makes sense to keep it then.